web/pgadmin/static/js/helpers/ModalProvider.jsx

```diff
@@ -53,18 +54,20 @@ function renderExtraButtons(button) {
 }
 
 function AlertContent({ text, confirm, okLabel = gettext('OK'), cancelLabel = gettext('Cancel'), onOkClick, onCancelClick, extraButtons }) {
+  // Check if any extra buttons have an focus property to determine if autofocus should be applied.
+  const hasAutoFocus = _.some(extraButtons, item => !_.isNil(item.focus));
   return (
     <StyledBox display="flex" flexDirection="column" height="100%">
       <Box flexGrow="1" p={2}>{typeof (text) == 'string' ? HTMLReactParser(text) : text}</Box>
       <Box className='Alert-footer'>
         {confirm &&
-          <DefaultButton startIcon={<CloseIcon />} onClick={onCancelClick} autoFocus={true}>{cancelLabel}</DefaultButton>
+          <DefaultButton startIcon={<CloseIcon />} onClick={onCancelClick} autoFocus={hasAutoFocus} >{cancelLabel}</DefaultButton>
         }
         {
           extraButtons?.length ?
             extraButtons.map(button=>renderExtraButtons(button))
             :
-            <PrimaryButton className='Alert-margin' startIcon={<CheckRoundedIcon />} onClick={onOkClick}>{okLabel}</PrimaryButton>
+            <PrimaryButton className='Alert-margin' startIcon={<CheckRoundedIcon/>} onClick={onOkClick} autoFocus={true}>{okLabel}</PrimaryButton>
```

Comment:
You have applied autofocus to this button and above Default button also. this will not work.

Reply:
As per this requirement :- 
**For the delete option, I would also:
- Make Cancel the default option.**

I added Auto-focus true on the Cancel button when there Is an extra button with a focus prop. that's why I've added conditionally Auto-focus at that place

Reply:
And if there are no extra buttons then? hasFocus will be true and default button + primary button both will have autoFocus.

Reply:
The focus does not depend on the extra button. It depends on the focus property of the extra button. And this focus property we are passing only for **Delete** button.

Reply:
As per your logic, if any extra has focus then default button will have focus?

Reply:
![Uploading image.png…]()

Reply:
As we are checking focus props from extraButtons and
If there are extraButtons then this **PrimaryButton** won't render.
It will render a button from **renderExtraButtons**.

Reply:
I remember when extraButtons was added, intention was to show `extra` buttons apart from existing ones. Now you've changed it. Extras is no more extra. It has become the main buttons.